modules/core/src/main/java/org/apache/ignite/internal/processors/query/GridQueryProcessor.java

```diff
@@ -559,6 +559,20 @@ else if (failOnStaticCacheSchemaChanges(cacheDesc)) {
                         msg.deploymentId(cacheDesc.deploymentId());
 
                     assert F.eq(cacheDesc.deploymentId(), msg.deploymentId());
+
```

Comment:
Does this change relate to the original ticket? I'm not against the fix though. )))
Perhaps it worth creating a new ticket to address this issue. Anyway, it is up to you )

Reply:
If I do not fix it, TC will fail on some tests.